src/algos/base_class.py

```diff
@@ -575,9 +589,46 @@ def local_train(self, round: int, epochs: int = 1, **kwargs: Any) -> Tuple[float
         if self.is_working:
             avg_loss, avg_acc = 0, 0
             for _ in range(epochs):
-                tr_loss, tr_acc = self.model_utils.train(
-                    self.model, self.optim, self.dloader, self.loss_fn, self.device, malicious_type=self.config.get("malicious_type", "normal"), config=self.config,
-                )            
+                if self.config.get("use_dpsgd"):
+                    self.log_utils.log_console(
+                        "Training with DPSGD"
+                    )
+                    epsilon = self.config.get("epsilon")
+                    if epsilon:
```

Comment:
I think this implementation will give incorrect privacy guarantee because for every round we will create a new privacy engine which leads to underestimation of epsilon.